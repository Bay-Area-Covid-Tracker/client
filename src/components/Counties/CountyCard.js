import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
	root: { width: 300 },
	title: { fontSize: 14 }
});

export default function CountyCard({ county }) {
	const classes = useStyles();

	return (
		<Card className={classes.root}>
			<CardContent>
				<Typography variant="h5">
					{county.county}
				</Typography>
			</CardContent>
		</Card>
	);
}

CountyCard.propTypes = { county: PropTypes.object };
CountyCard.defaultProps = {
	county: {
		id: 0,
		county: '',
		state: '',
		country: 'US',
		lastUpdate: '',
		lat: 0.000,
		lon: 0.000,
		confirmed: 0,
		deaths: 0
	}
};
