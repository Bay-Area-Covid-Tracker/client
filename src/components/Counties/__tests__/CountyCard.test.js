import React from 'react';
import { shallow, mount } from 'enzyme';

import CountyCard from '../CountyCard';

const county = {
	"id": 6055,
	"county": "Napa",
	"state": "California",
	"country": "US",
	"lastUpdate": "2020-03-30 22:52:45",
	"lat": 38.5074,
	"lon": -122.333,
	"confirmed": 14,
	"deaths": 0
};

describe('tests for CountyCard', () => {
	describe('snapshot tests', () => {
		it('renders without crashing', () => {
			const tree = shallow(<CountyCard />);
			expect(tree.html()).toMatchSnapshot();
		});
	});

	describe('unit tests', () => {
		it('renders with passed county prop', () => {
			const wrapper = mount(<CountyCard county={county} />);
			expect(wrapper.props().county).toEqual(county);
		});
	});
});
