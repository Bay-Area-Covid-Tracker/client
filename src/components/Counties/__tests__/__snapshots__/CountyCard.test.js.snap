// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for CountyCard snapshot tests renders without crashing 1`] = `"<div class=\\"MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation4 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\"></h5><br/><p class=\\"MuiTypography-root MuiTypography-body1\\">Confirmed</p><h6 class=\\"MuiTypography-root MuiTypography-h6\\">0</h6><p class=\\"MuiTypography-root MuiTypography-body1\\">Deaths</p><h6 class=\\"MuiTypography-root MuiTypography-h6\\">0</h6></div></div>"`;
